### PR TITLE
Make AddFollower and RemoveFollower less aggressive.

### PR DESCRIPTION
**Pull request by neunhoef**

### CODE REVIEW

File: arangod/Agency/Job.cpp [hunk 1]
```diff
@@ -228,7 +228,7 @@ std::string Job::randomIdleAvailableServer(Node const& snap, Slice const& exclud
 
 // The following counts in a given server list how many of the servers are
 // in Status "GOOD".
```
Comment (maierlars) [inline arangod/Agency/Job.cpp]:
```suggestion
 // in Status "GOOD" or "BAD".
```

File: arangod/Agency/Job.cpp [hunk 2]
```diff
@@ -260,7 +261,7 @@ size_t Job::countGoodServersInList(Node const& snap, VPackSlice const& serverLis
...
```
Comment (maierlars) [inline arangod/Agency/Job.cpp]:
```suggestion
 // in Status "GOOD" or "BAD".
```

File: arangod/Agency/Supervision.cpp [hunk 3]
```diff
@@ -1258,6 +1258,26 @@ void Supervision::readyOrphanedIndexCreations() {
...
   _lock.assertLockedByCurrentThread();
+
+  // First check the number of AddFollower and RemoveFollower jobs in ToDo:
+  auto todos = _snapshot.hasAsChildren(toDoPrefix).first;
+  int nrAddRemoveJobsInTodo = 0;
+  for (auto it = todos.begin(); it != todos.end(); ++it) {
+    auto jobNode = *(it->second);
+    auto t = jobNode.hasAsString("type");
+    if (t.second && (t.first == "addFollower" || t.first == "removeFollower")) {
+      ++nrAddRemoveJobsInTodo;
```
Comment (maierlars) [inline arangod/Agency/Supervision.cpp]: Maybe we should define a `maxAddRemoveJobsInTodo` and then early exit here if `nrAddRemoveJobsInTodo` is already bigger.

Comment (maierlars) [reply arangod/Agency/Supervision.cpp]: Or return immediately...

Comment (neunhoef) [reply arangod/Agency/Supervision.cpp]: Done

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3579/

**Pull request merged**